packages/dev/core/src/Engines/Extensions/engine.dynamicBuffer.ts

```diff
@@ -30,10 +30,13 @@ ThinEngine.prototype.updateDynamicIndexBuffer = function (this: ThinEngine, inde
     this.bindIndexBuffer(indexBuffer);
     let arrayBuffer;
 
-    if (indices instanceof Uint16Array || indices instanceof Uint32Array) {
-        arrayBuffer = indices;
-    } else {
-        arrayBuffer = indexBuffer.is32Bits ? new Uint32Array(indices) : new Uint16Array(indices);
+    if (indexBuffer.is32Bits) {
+        // anything else than Uint32Array needs to be converted to Uint32Array
+        arrayBuffer = indices instanceof Uint32Array ? indices : Uint32Array.from(indices)
+    }
+    else {
+        // anything else than Uint16Array needs to be converted to Uint16Array
+        arrayBuffer = indices instanceof Uint16Array ? indices : Uint16Array.from(indices)

```

Comment:
And by the way, IMHO, thing are easier to understand if written this way.

Another example is in `WebGPUEngine.updateDynamicIndexBuffer` :
```
let view: ArrayBufferView;
if (indices instanceof Uint16Array) {
    if (indexBuffer.is32Bits) {
        view = Uint32Array.from(indices);
    } else {
        view = indices;
    }
} else if (indices instanceof Uint32Array) {
    if (indexBuffer.is32Bits) {
        view = indices;
    } else {
        view = Uint16Array.from(indices);
    }
} else {
    if (indexBuffer.is32Bits) {
        view = new Uint32Array(indices);
    } else {
        view = new Uint16Array(indices);
    }
}
```

might be clearer written this way as well:
```
    let view: ArrayBufferView;
    if (indexBuffer.is32Bits) {
        // anything else than Uint32Array needs to be converted to Uint32Array
        view = indices instanceof Uint32Array ? indices : Uint32Array.from(indices)
    }
    else {
        // anything else than Uint16Array needs to be converted to Uint16Array
        view = indices instanceof Uint16Array ? indices : Uint16Array.from(indices)
    }
```